### PR TITLE
feat(sc-57427): Support Opportunities API

### PR DESCRIPTION
**Pull request by kamilpavlicko**
closes ticket: https://app.shortcut.com/chartmogul/story/57427/support-opportunities-resource-ruby-client?vc_group_by=day&ct_workflow=all&cf_workflow=500105879

API document: https://www.notion.so/chartmogul/API-endpoints-for-Opportunities-a92323514bf44f89ad2f97ae30cee7f0#6f09cc7a71bc40e69ef65d8d0342d230 

### Background

1. Create a opportunity 
```
customer = ChartMogul::Customer.retrieve('cus_3819e09a-50a2-11ee-ada7-9fcf71cd4580')
customer.create_opportunity(
  ....
)

ChartMogul::Opportunity.create(data)
```

2. Get all opportunities
```
customer = ChartMogul::Customer.retrieve('cus_3819e09a-50a2-11ee-ada7-9fcf71cd4580')
customer.opportunities

ChartMogul::Opportunity.all(customer_uuid: customer_uuid)
```

3. Get an opportunity

```
opportunities = ChartMogul::Opportunity.retrieve('opportunity_uuid')
```

4. Update an opportunity
```
updated_opportunity = ChartMogul::Opportunity.update(data)
```

5. Delete an opportunity
```
ChartMogul::Opportunity.delete('opportunity_uuid') 
```

### CODE REVIEW

**Review by tudorpavel — APPROVED** [review]

File: spec/chartmogul/opportunity_spec.rb [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    subject { described_class.new(attrs) }
+
+    it 'sets the read-only properties correctly' do
+      expect(subject).to have_attributes({ uuid: nil })
+    end
+
+    it 'sets the writeable properties correctly' do
+      expect(subject).to have_attributes(attrs.reject { |k, _| k == :uuid })
+    end
+  end
```
Comment (SoeunSona) [inline spec/chartmogul/opportunity_spec.rb:38]: [minor] we can add more tests for `create_at` and `updated_at`

File: spec/chartmogul/opportunity_spec.rb [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    context 'with pagination' do
+      let(:first_cursor) do
+        'MjAyNC0wMy0xMVQwNzo0NTo0MS4xMzM4NjEwMDBaJjViZWUwYTQyLWRmN2ItMTFlZS05NTA2LWNiMjM4ZTliMTIxOA=='
+      end
+      let(:next_cursor) do
+        'MjAyNC0wMy0xMVQwNjozODo0Ny44Nzk2NDYwMDBaJjAzZDdkNWJjLWRmNzItMTFlZS1iMTBkLThiNzg5ZDc1N2MyYQ=='
+      end
+
+      it 'paginates correctly' do
+        opportunities = ChartMogul::Opportunity.all(per_page: 1)
```
Comment (SoeunSona) [inline spec/chartmogul/opportunity_spec.rb:108]: [question] I'm curious if we don't need to add `customer_uuid` to retrieve all opportunity

**Review by SoeunSona — APPROVED** [review]

File: spec/chartmogul/opportunity_spec.rb (hunk 2 above)
Comment (kamilpavlicko) [reply spec/chartmogul/opportunity_spec.rb:108]: good question, `customer_uuid` is optional in this case 👍

File: spec/chartmogul/opportunity_spec.rb (hunk 1 above)
Comment (kamilpavlicko) [reply spec/chartmogul/opportunity_spec.rb]: added 👍

**Review by SoeunSona — APPROVED** [review]

**Pull request merged**